pkg/server/server.go

```diff
@@ -287,12 +284,149 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	})
 }
 
-func renderTemplate(template string, identity *token.Identity) string {
+func (h *handler) doMapping(identity *token.Identity, arn string) (string, []string, error) {
+	if roleMapping, exists := h.lowercaseRoleMap[arn]; exists {
+		username, err := h.renderTemplate(roleMapping.Username, identity)
+		if err != nil {
+			return "", nil, fmt.Errorf("failed mapping username: %s", err.Error())
+		}
+		groups := []string{}
+		for _, groupPattern := range roleMapping.Groups {
+			group, err := h.renderTemplate(groupPattern, identity)
+			if err != nil {
+				return "", nil, fmt.Errorf("failed mapping group: %s", err.Error())
+			}
+			groups = append(groups, group)
+		}
+		return username, groups, nil
+	} else if userMapping, exists := h.lowercaseUserMap[arn]; exists {
```

Comment:
Nit: these `else if` clauses could now be plain `if` clauses since we return early at the end of each one.

Reply:
Done.